test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -607,7 +607,35 @@ public NodeToolResult nodetoolResult(boolean withNotifications, String... comman
             try (CapturingOutput output = new CapturingOutput())
             {
                 DTestNodeTool nodetool = new DTestNodeTool(withNotifications, output.delegate);
-                int rc = nodetool.execute(commandAndArgs);
+                // install security manager to get informed about the exit-code
+                System.setSecurityManager(new SecurityManager()
```

Comment:
Indeed an interesting approach to intercept `System.exit` using security manager!

Reply:
took from ToolRunner =D